File: validation/VariableGenerator.py
# -*- coding: utf-8 -*-
import itertools
from enum import Enum

i = itertools.count()


class VariableGenerator:
    def __init__(self):
        self.index = 0

    #def incrementAndGet(self):
    #    self.index += 1
    #    return self.index

    @staticmethod
    def generateVariable(type):
        type = "p_"  # *** hardcoded
        return str(type) + str(next(i))

    @staticmethod
    def getFocusNodeVar():
        return "x"  # ***


class VariableType(Enum):
    VALIDATION = "p_"
    VIOLATION = "n_"  # not used

    @classmethod
    def prefix(cls):
        return cls.VALIDATION

File: engines/shacl2sparql/src/main/java/unibz/shapes/core/global/VariableGenerator.java
package unibz.shapes.core.global;


import java.util.concurrent.atomic.AtomicInteger;

public class VariableGenerator {

    public enum VariableType {
        VALIDATION("p_"),
        VIOLATION("n_");
        public final String prefix;

        VariableType(String prefix) {
            this.prefix = prefix;
        }
    }
        private static final AtomicInteger index = new AtomicInteger(0);

        public static String generateVariable(VariableType type) {
            return type.prefix + index.incrementAndGet();
        }

        public static String getFocusNodeVar() {
            return "x";
        }

}
